pyodide-build/pyodide_build/config.py

```diff
@@ -27,7 +28,7 @@ def __init__(self, pyodide_root: Path):
         self._config = {
             **self._load_default_config(),
             **self._load_makefile_envs(),
-            **self._load_config_file(),
+            **self._load_config_file(pyodide_root / "pyproject.toml", os.environ),
```

Comment:
Let's not read the configuration file from pyodide_root. In out-of-tree build, normally users will not access that directory.

Instead, let's pass the cwd as a parameter, and search the pyproject.toml file by looking for the parent directories (just like how we do in [search_pyodide_root](https://github.com/pyodide/pyodide/blob/ff3942e942cda1e538a6f83247ebfb0f93cecbd5/pyodide-build/pyodide_build/build_env.py#L94C5-L94C24))

Reply:
might be a bit too much of a shortcut, but I've changed it to directly use `search_pyodide_root` here instead to achieve the same thing. let me know if it feels wrong 🤔

Reply:
You should not use `search_pyodide_root,` but I think we can extract some common parts in `search_pyodide_root` and use them here.

When it comes to building packages, pyodide-build has two modes: in-tree and out-of-tree. the in-tree build is when building packages inside `pyodide/pyodide` repository, out-of-tree build is when building packages somewhere else.

`search_pyodide_root` is used to detect if we are in in-tree build, while this PR targets both in-tree and out-of-tree builds. So, using search_pyodide_root here is not what we want.

Instead, I think we can extract a few parts of `search_pyodide_root` and make it a separate function: searching `pyproject.toml` by iterating the parent directories, and returning the content of pyproject.toml. Then we can use that function in both search_pyodide_root and here.